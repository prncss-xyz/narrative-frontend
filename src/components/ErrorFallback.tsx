import { Box } from "./basics";

interface Error {
  message: string;
}

export function ErrorFallback({ error }: { error: Error }) {
  // TODO: could send log to server
  console.error(error);
  return (
    <Box role="alert">
      <Box fontSize={3}>Something went wrong.</Box>
      <pre>{error.message}</pre>
    </Box>
  );
}
