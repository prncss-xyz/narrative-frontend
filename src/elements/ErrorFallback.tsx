import { Box } from "./basics";

export function ErrorFallback({ error }: { error: any }) {
  // TODO: could send log to server
  console.error(error);
  return (
    <Box role="alert">
      <Box fontSize={3}>Something went wrong.</Box>
      <pre>{error?.message}</pre>
    </Box>
  );
}
