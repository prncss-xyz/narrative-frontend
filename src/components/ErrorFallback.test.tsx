import { render } from "@testing-library/react";
import { ErrorFallback } from "./ErrorFallback";

describe("ErrorFallback", () => {
  it("should render correctly", () => {
    const { container } = render(
      <ErrorFallback error={{ message: "message" }} />
    );
    expect(container).toMatchSnapshot();
  });
});
