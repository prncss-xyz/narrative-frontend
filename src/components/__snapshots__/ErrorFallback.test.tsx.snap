// Vitest Snapshot v1

exports[`ErrorFallback > should render correctly 1`] = `
<div>
  <div
    class="css-1x3sxzl-Box e2i8k6n3"
    role="alert"
  >
    <div
      class="css-wn5gmn-Box e2i8k6n3"
      font-size="3"
    >
      Something went wrong.
    </div>
    <pre>
      message
    </pre>
  </div>
</div>
`;
